src/mqueryfront/src/RecentPage.js

```diff
@@ -1,17 +1,135 @@
 import React, { Component } from "react";
 import SearchJobs from "./SearchJobs";
+import ErrorBoundary from "./ErrorBoundary";
+import axios from "axios";
+import { API_URL } from "./config";
 
 class RecentPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            jobs: [],
+            head: [],
+            filter: null,
+            error: null,
+        };
+
+        this.handleCancelJob = this.handleCancelJob.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.handleFilter = this.handleFilter.bind(this);
+        this._getHead = this._getHead.bind(this);
```

Comment:
Why underscored? AFAIK underscores in javascript are pretty polarising (i.e. not everyone likes them). Also I think they're not more private than handleClose for example. So maybe they should just be called `getHead` and `getDistinctList`?

I don't have a strong opinion here, just know that some JS style guides disallow underscores in field names

Reply:
JS do not support private methods. I used pseudo privacy by marking "private" method with underscore. See: https://stackoverflow.com/questions/27849064/how-to-implement-private-method-in-es6-class-with-traceur

Reply:
Yes, OTOH all methods here are technically private (`handleClose` is private too, because external components are not supposed to call it).

I ask because it's not very common in react code. Some style guides (like AirBNB) even explicitly disallow them: https://github.com/airbnb/javascript/tree/master/react#methods.

I'm not decided either way, but we don't use this convention anywhere else so not sure about this.